src/catalog/src/builtin.rs

```diff
@@ -3245,8 +3245,33 @@ pub const MZ_COMPUTE_ERROR_COUNTS_PER_WORKER: BuiltinView = BuiltinView {
     name: "mz_compute_error_counts_per_worker",
     schema: MZ_INTERNAL_SCHEMA,
     sql: "CREATE VIEW mz_internal.mz_compute_error_counts_per_worker AS
-SELECT export_id, worker_id, count
-FROM mz_internal.mz_compute_error_counts_raw",
+WITH
+    -- Indexes that reuse existing indexes rather than maintaining separate dataflows.
+    -- For these we don't log error counts separately, so we need to forward the error counts from
+    -- their dependencies instead.
+    index_reuses AS (
+        SELECT d.object_id, d.dependency_id
+        FROM mz_internal.mz_compute_dependencies d
+        JOIN mz_internal.mz_compute_exports e
+            ON (e.export_id = d.object_id)
+        WHERE NOT EXISTS (
+            SELECT 1 FROM mz_internal.mz_dataflows
+            WHERE id = e.dataflow_id
+        )
+    ),
+    -- Error counts that were directly logged on compute exports.
+    direct_errors AS (
+        SELECT export_id, worker_id, count
+        FROM mz_internal.mz_compute_error_counts_raw
+    ),
+    -- Error counts that were not directly logged but forwarded from reused indexes.
+    indirect_errors AS (
+        SELECT r.object_id, e.worker_id, e.count
+        FROM index_reuses r
+        JOIN direct_errors e
+            ON (r.dependency_id = e.export_id)
+    )
+SELECT * FROM direct_errors UNION ALL (SELECT * FROM indirect_errors)",
```

Comment:
It's great to have the error count based on the transitive closure, but I have two concerns:
1. Does `index_reuses` recursively show indexes? I don't have the semantics of `mz_compute_dependencies/exports` in my head, so this might be the case already! I think we need to use the transitive closure here.
2. It feels this should be a separate view, because otherwise it might be more difficult to pin where errors come from, but I don't have a strong opinion.

Reply:
1. No, it's not recursive. This view is based on the assumption that indexes are never transitively reused. I.e. if there is one dataflow producing an arrangements, copies of that dataflow will all depend on the first one, not on each other. I only did a cursory test to verify this, which is not enough. I'll verify in the code.
2. Hm, I would like to avoid users having to know about the fact that we reuse indexes. I don't think this makes it more difficult to find errors either! If we report an error in a reused index, you can look at our various dependency views to check if any dependencies have errors. Or you can look at the SQL of the view's index view `SHOW CREATE`, or you can look at the `EXPLAIN` for that index. Each of these work the same for reused and non-reused indexes.

Reply:
> This view is based on the assumption that indexes are never transitively reused.

AFAICT that's only the case today because index imports get sorted by the index ID so the oldest index is always the first and that's the one that gets then picked. This is an implementation detail we should rely on, so I made this view check transitive reuses instead.